weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableColumns.tsx

```diff
@@ -637,3 +510,146 @@ const isExpandedRefWithValueAsTableRef = (
+  onExpand?: (col: string) => void,
+  columnIsFilterable?: (col: string) => boolean,
+  columnIsSortable?: (col: string) => boolean
+) => {
+  const cols: Array<GridColDef<T>> = [];
+
+  const tree = buildTree([...filteredDynamicColumnNames]);
+  let groupingModel: GridColumnGroupingModel = tree.children.filter(
+    c => 'groupId' in c
+  ) as GridColumnGroup[];
+
+  const walkGroupingModel = (
+    nodes: GridColumnNode[],
+    fn: (node: GridColumnNode) => GridColumnNode
+  ) => {
+    return nodes.map(node => {
+      node = fn(node);
+      if ('children' in node) {
+        node.children = walkGroupingModel(node.children, fn);
+      }
+      return node;
+    });
+  };
+  const groupIds = new Set<string>();
+  groupingModel = walkGroupingModel(groupingModel, node => {
+    if ('groupId' in node) {
+      const key = node.groupId;
+      groupIds.add(key);
+      if (columnIsExpanded && onCollapse && columnIsExpanded(key)) {
+        node.renderHeaderGroup = () => {
+          return (
+            <CollapseHeader
+              headerName={key.split('.').slice(-1)[0]}
+              field={key}
+              onCollapse={onCollapse}
+            />
+          );
+        };
+      } else if (columnCanBeExpanded && onExpand && columnCanBeExpanded(key)) {
+        node.renderHeaderGroup = () => {
+          return (
+            <ExpandHeader
+              headerName={key.split('.').slice(-1)[0]}
+              field={key}
+              hasExpand
+              onExpand={onExpand}
+            />
+          );
+        };
+      }
+    }
+    return node;
+  }) as GridColumnGroupingModel;
+
+  for (const key of filteredDynamicColumnNames) {
+    const col: GridColDef<T> = {
+      flex: 1,
+      minWidth: 150,
+      field: key,
+      filterable: columnIsFilterable && columnIsFilterable(key),
+      sortable: columnIsSortable && columnIsSortable(key),
+      filterOperators: allOperators,
+      headerName: key,
+      renderHeader: () => {
+        return (
+          <div
+            style={{
+              fontWeight: 600,
+            }}>
+            {key.split('.').slice(-1)[0]}
+          </div>
+        );
+      },
+      valueGetter: cellParams => {
+        const val = valueForKey(cellParams.row, key);
+        if (Array.isArray(val) || typeof val === 'object') {
+          try {
+            return JSON.stringify(val);
+          } catch {
+            return val;
+          }
+        }
+        return val;
+      },
+      renderCell: cellParams => {
+        const val = valueForKey(cellParams.row, key);
+        if (val === undefined) {
+          return <NotApplicable />;
+        }
+        return (
+          <ErrorBoundary>
+            {/* In the future, we may want to move this isExpandedRefWithValueAsTableRef condition
+            into `CallValue`. However, at the moment, `ExpandedRefWithValueAsTableRef` is a
+            CallsTable-specific data structure and we might not want to leak that into the
```

Comment:
Is this comment outdated, no longer CallsTable-specific?

Reply:
fixed